### PR TITLE
[FLINK-10900][kafka][docs] Mark universal Kafka connector as beta

### PR DESCRIPTION
**Pull request by pnowojski**
Because of frequent deadlocks and other failures in tests the connector is marked as beta. Those errors are probably only test issues or Kafka bugs but as for now it's not confirmed.

## Verifying this change

This is a change in documentation only.

### CODE REVIEW

**Comment by pnowojski** [comment]
CC @tzulitai @yanghua

File: docs/dev/connectors/kafka.md [hunk 1]
```diff
@@ -73,21 +73,33 @@ For most users, the `FlinkKafkaConsumer08` (part of `flink-connector-kafka`) is
...
         FlinkKafkaProducer</td>
         <td>>= 1.0.0</td>
-        <td>This Kafka connector attempts to track the latest version of the Kafka client. The version of the client it uses may change between Flink releases. Modern Kafka clients are backwards compatible with broker versions 0.10.0 or later. However for Kafka 0.11.x and 0.10.x versions, we recommend using dedicated flink-connector-kafka-0.11 and link-connector-kafka-0.10 respectively.</td>
-        </tr>
+        <td>
+        This universal Kafka connector attempts to track the latest version of the Kafka client.
+        The version of the client it uses may change between Flink releases.
+        Modern Kafka clients are backwards compatible with broker versions 0.10.0 or later.
+        However for Kafka 0.11.x and 0.10.x versions, we recommend using dedicated
+        flink-connector-kafka-0.11 and link-connector-kafka-0.10 respectively.
```
Comment (yanghua) [inline docs/dev/connectors/kafka.md]: Shall we also add `{{ site.scala_version_suffix }}` suffix here?

File: docs/dev/connectors/kafka.md [hunk 2]
```diff
@@ -73,21 +73,33 @@ For most users, the `FlinkKafkaConsumer08` (part of `flink-connector-kafka`) is
...
+        <div class="alert alert-warning">
+          <strong>Attention:</strong> as of Flink 1.7 the universal Kafka connector is considered to be
+          in a <strong>BETA</strong> status and might not be as stable as the 0.11 connector.
+          In case of problems with the universal connector, you can try to use flink-connector-kafka-0.11{{ site.scala_version_suffix }}
+          which should be compatible with all of the Kafka versions starting from 0.11.
```
Comment (yanghua) [inline docs/dev/connectors/kafka.md]: Here, maybe not. Do you remember the producer transaction uses reflect to access the inner private field ([sequenceNumbers 0.11 -> nextSequence 1.0+](https://github.com/apache/kafka/commit/5d2422258cb975a137a42a4e08f03573c49a387e#diff-f4ef1afd8792cd2a2e9069cd7ddea630)) ?

File: docs/dev/connectors/kafka.md [hunk 3]
```diff
@@ -110,13 +123,17 @@ Note that the streaming connectors are currently not part of the binary distribu
...
+Rather, it tracks the latest version of Kafka at the time of the Flink release.
 
-If your Kafka broker version is 1.0.0 or newer, you should use this Kafka connector. If you use an older version of Kafka (0.11, 0.10, 0.9, or 0.8), you should use the connector corresponding to the broker version.
+If your Kafka broker version is 1.0.0 or newer, you should use this Kafka connector.
+If you use an older version of Kafka (0.11, 0.10, 0.9, or 0.8), you should use the connector corresponding to the broker version.
 
 ### Compatibility
 
-The modern Kafka connector is compatible with older and newer Kafka brokers through the compatibility guarantees of the Kafka client API and broker. The modern Kafka connector is compatible with broker versions 0.11.0 or later, depending on the features used. For details on Kafka compatibility, please refer to the [Kafka documentation](https://kafka.apache.org/protocol.html#protocol_compatibility).
+The modern Kafka connector is compatible with older and newer Kafka brokers through the compatibility guarantees of the Kafka client API and broker.
```
Comment (yanghua) [inline docs/dev/connectors/kafka.md]: I think we should give this connector a consistent name, either `universal ` or `modern`. Not both.

File: docs/dev/connectors/kafka.md (hunk 2 above)
Comment (pnowojski) [reply docs/dev/connectors/kafka.md]: This is a good point, but as far as I know this is related only if someone would want to upgrade `kafka.version` of the `flink-connector-kafka`. There were couple of users that were successfully using `flink-connector-kafka-0.11` against Kafka 1.0 and it should work for the same reasons why `universal` connector should work with any Kafka version `0.11+`

File: docs/dev/connectors/kafka.md (hunk 3 above)
Comment (pnowojski) [reply docs/dev/connectors/kafka.md]: I was just braking new lines here, but I can fix this while I'm at it. Thanks for pointing out.

**Pull request merged**